agent/agent.go

```diff
@@ -103,8 +103,14 @@ func NewAgent(
 	metricsReporters []metrics.Reporter,
 ) *Agent {
 	// initialize heartbeat and handshake data on first user connection
+	serializerName := ""
+
+	if serializer != nil { //should never be true, only during testing
```

Comment:
I don't like this approach. IMO we'd better ensure serializer is not `nil` on the tests causing this issue instead of inserting this clause.

Reply:
I was lazy when coding this, I also prefer this other approach!